src/Week2/LogAnalysis.hs

```diff
@@ -8,41 +8,33 @@ import Week2.Log
 
 parseError :: String -> LogMessage
 parseError string =
-        readError $ words string
+    case words string of
+        "I":n:xs -> maybe (Unknown string) (makeLog Info xs) (readMaybe n :: Maybe Int)
```

Comment:
You don't need to write type of `readMaybe` here explicitly because it's inferred from the field of `LogMessage`